### PR TITLE
Added CreateDeleteResponse to ResultHandler, Refactored GET, POST and DELETE response methods

### PR DESCRIPTION
**Pull request by PurpleBriar**
# Type of PR

- [ ] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] My code follows the code style of this project.
- [x] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

- Added CreateDeleteResponse to ResultHandler
- Refactored GET, POST and DELETE response methods

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [x] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced
References #https://github.com/retaildevcrews/wcnp/issues/581

### CODE REVIEW

File: src/LodeRunner.API/src/Handlers/ResultHandler.cs [hunk 1]
```diff
@@ -231,8 +213,53 @@ public static async Task<ActionResult> CreatePostResponse<TEntity>(Func<TEntity,
                 logger.LogError(new EventId((int)HttpStatusCode.InternalServerError, $"{methodName} > {nameof(CreatePutResponse)}"), ex, "Exception");
```
Comment (gortegaMS) [inline src/LodeRunner.API/src/Handlers/ResultHandler.cs:213]: Should we have local variable here as well?

**Review by gortegaMS — CHANGES_REQUESTED** [review]

File: src/LodeRunner.API/src/Controllers/TestRunsController.cs [hunk 2]
```diff
@@ -190,55 +190,61 @@ public async Task<ActionResult> DeleteTestRun([FromRoute, SwaggerRequestBody("Th
...
                 return service.Validator.ValidateEntity(testRun);
             });
         }
+
+        /// <summary>
+        /// Check if conditions are met for testRun deletion.
+        /// </summary>
+        /// <param name="testRun">TestRun entity.</param>
+        /// <returns>Boolean value indicating whether testRun can be deleted.</returns>
+        private bool CanTestRunBeDeleted(TestRun testRun)
```
[the remark this answers is not in the document]
Comment (PurpleBriar) [reply src/LodeRunner.API/src/Controllers/TestRunsController.cs:216]: Updated to a static method

File: src/LodeRunner.API/src/Handlers/ResultHandler.cs [hunk 3]
```diff
@@ -141,8 +137,8 @@ public static async Task<ActionResult> CreateGetByIdResponse<TEntity>(Func<strin
         /// <returns>A task with the appropriate response.</returns>
         public static async Task<ActionResult> CreatePostResponse<TEntity>(Func<TEntity, CancellationToken, Task<TEntity>> postResult, TEntity payload, string path, IEnumerable<string> errorList, ILogger logger, CancellationToken cancellationToken, [CallerMemberName] string methodName = null)
         {
-            try
-            {
+            return await TryCatchException(logger, $"{methodName} > {nameof(CreateGetByIdResponse)}", async () =>
```
[the remark this answers is not in the document]
Comment (PurpleBriar) [reply src/LodeRunner.API/src/Handlers/ResultHandler.cs:140]: Added a local variable

**Review by jofultz — CHANGES_REQUESTED** [review]
Checks failed.

**Review by gortegaMS — APPROVED** [review]

**Review by jofultz — APPROVED** [review]

**Review by joheec — APPROVED** [review]

**Pull request merged**